featuretools/entityset/serialization.py

```diff
@@ -1,175 +1,228 @@
+    with open(file, 'w') as file:
+        json.dump(description, file)
+
+
+def from_variable_description(description):
+    '''Deserialize variable from variable description.
+
+    Args:
+        description (dict) : Description of :class:`.Variable`.
+
+    Returns:
+        variable (Variable) : Returns :class:`.Variable`.
+    '''
+    is_string = isinstance(description['type'], str)
+    type = description['type'] if is_string else description['type']['value']
+    return VARIABLE_TYPES.get(type, VARIABLE_TYPES.get(None))
+
+
+def get_variable_types(description):
+    '''Returns variable types from entity description.
+
+    Args:
+        description (dict) : Description of :class:`.Entity`.
 
+    Returns:
+        variable_types (dict) : Returns dictionary where variable id maps to variable type.
+    '''
+    return {variable['id']: from_variable_description(variable) for variable in description['variables']}
+
+
+def from_entity_description(entityset, description, path=None):
+    '''Deserialize entity from entity description and add to entityset.
+
+    Args:
+        entityset (EntitySet) : Instance of :class:`.EntitySet` to add :class:`.Entity`.
+        description (dict) : Description of :class:`.Entity`.
+        path (str) : Root directory to serialized entityset.
+    '''
+    dataframe = read_entity_data(description, path=path)
+    variable_types = get_variable_types(description)
+    dtypes = description['loading_info']['properties']['dtypes']
+    entityset.entity_from_dataframe(
+        description['id'],
+        dataframe.astype(dtypes),
+        index=description.get('index'),
+        time_index=description.get('time_index'),
+        secondary_time_index=description['properties'].get('secondary_time_index'),
+        variable_types=variable_types)
+
+
+def from_relationship_description(entityset, description):
+    '''Deserialize parent and child variables from relationship description.
+
+    Args:
+        entityset (EntitySet) : Instance of :class:`.EntitySet` containing parent and child variables.
+        description (dict) : Description of :class:`.Relationship`.
+
+    Returns:
+        item (tuple(Variable, Variable)) : Tuple containing parent and child variables.
+    '''
+    entity, variable = description['parent']
+    parent = entityset[entity][variable]
+    entity, variable = description['child']
+    child = entityset[entity][variable]
+    return parent, child
+
+
+def read_entity_data(description, path=None):
+    '''Read description data from disk.
 
-def _write_parquet_entity_data(root, entity, metadata,
-                               engine='auto', compression='gzip'):
-    '''Write an Entity's data to the binary parquet format, using pd.DataFrame.to_parquet.
+    Args:
+        description (dict) : Description of :class:`.Entity`.
+
+    Returns:
+        df (DataFrame) : Instance of dataframe. Returns an empty dataframe path is not specified.
+    '''
+    if path is None:
```

Comment:
I think we should do this logic for handling `path` in `from_entity_description` and only call `read_entity_data` if path is not None. it's a bit unintuitive that you can call a method named `read` without an reading happening

Reply:
Change applied in [this commit.](https://github.com/Featuretools/featuretools/pull/361/commits/a9a6a4a08061d4c96789aa8065349273e76c5ebe)